src/libraries/AccountAccessParser.sol

```diff
@@ -77,86 +115,207 @@ library StateDiffDecoder {
     bytes32 internal constant GAS_PAYING_TOKEN_SYMBOL_SLOT = bytes32(uint256(keccak256("opstack.gaspayingtokensymbol")) - 1);
     // forgefmt: disable-end
 
-    function decode(VmSafe.AccountAccess[] memory _accountAccesses) internal {
-        vm.pauseGasMetering();
-        // First, determine the maximum possible number of storage writes.
-        uint256 maxEntries = 0;
-        for (uint256 i = 0; i < _accountAccesses.length; i++) {
-            for (uint256 j = 0; j < _accountAccesses[i].storageAccesses.length; j++) {
-                VmSafe.StorageAccess memory storageAccess = _accountAccesses[i].storageAccesses[j];
-                if (storageAccess.isWrite && storageAccess.previousValue != storageAccess.newValue) {
-                    maxEntries++;
-                }
+    modifier noGasMetering() {
+        // We use low-level staticcalls so we can keep cheatcodes as view functions.
+        (bool ok,) = address(vm).staticcall(abi.encodeWithSelector(VmSafe.pauseGasMetering.selector));
+        require(ok, "pauseGasMetering failed");
+
+        _;
+
+        (ok,) = address(vm).staticcall(abi.encodeWithSelector(VmSafe.resumeGasMetering.selector));
+        require(ok, "resumeGasMetering failed");
+    }
+
+    // ==============================================================
+    // ======== Methods intended to be used as the interface ========
+    // ==============================================================
+
+    /// @notice Convenience function that wraps decode and print together.
+    function decodeAndPrint(VmSafe.AccountAccess[] memory _accesses) internal view {
+        (DecodedTransfer[] memory transfers, DecodedStateDiff[] memory stateDiffs) = decode(_accesses);
+        print(transfers, stateDiffs);
+    }
+
+    /// @notice Decodes the provided AccountAccess array into decoded transfers and state diffs.
+    function decode(VmSafe.AccountAccess[] memory _accountAccesses)
```

Comment:
note to self: make sure to handle the `reverted` field properly for both account and storage accesses